### PR TITLE
Merge dev branch

### LINKED ISSUE #52: Enforce PEP8 standard and any docstring format
Code doesn't follow PEP8, and is lacking proper docstrings for public methods

### PR DESCRIPTION
**Pull request by delunalara**
Fixes #52, #45, #43, #30, #28, #21, #20, #17

### CODE REVIEW

File: emspy/connection.py [hunk 1]
```diff
@@ -1,184 +1,296 @@
...
+        verbose: bool
+            verbose output (default False)
+        ignore_ssl_errors: bool
+            ignore SSL certificate errors (default False)
+        server: str
+            server to connect to from emspy.common.uri_root
+        server_url: str
+            hostname for server to connect to if different from any of the common values
+        max_trials: int
+            maximum number of reconnection attempts
```
Comment (danverkamp) [inline emspy/connection.py:48]: Do we want to specify which parameters are optional?

File: emspy/query/fltquery.py [hunk 2]
```diff
@@ -1,627 +1,845 @@
...
+import json
+import sys
 from builtins import str
+from builtins import zip
 from collections import OrderedDict
-from emspy.query import *
-from .query import Query
 
 import pandas as pd
-import sys, json
```
Comment (danverkamp) [inline emspy/query/fltquery.py:11]: There's a lot to go through here with the PEP8 changes and whatever else was done. Can you summarize the non-PEP8 changes so I know where to look?

File: emspy/connection.py (hunk 1 above)
Comment (delunalara) [reply emspy/connection.py:48]: I think the class signature should make it evident which parameters are optional, but I can add the optional tags to all functions (would take me a while)

File: emspy/connection.py [hunk 3]
```diff
@@ -1,184 +1,296 @@
...
+        self.token = None
+        self.token_type = None
+
+        # We assign the uri root to a member variable up front, and use that everywhere to
+        # simplify. In order to use an alternate uri root, it must be specified in the constructor.
+        if server_url is not None:
+            if server_url.endswith('/api'):
+                self._uri_root = server_url
+            else:
+                self._uri_root = server_url.rstrip('/') + '/api'
```
[the remark this answers is not in the document]
Comment (delunalara) [reply emspy/connection.py:66]: Yeah, the changes in the first commit were only code styling

File: emspy/query/aircraft.py [hunk 4]
```diff
@@ -3,37 +3,82 @@
...
+            fleet id
+
+        Returns
+        -------
+        pd.DataFrame
+            aircrafts that match fleet id
+        """
+        assets = self.list_all()
+        idx = [row.id for row in assets.itertuples() if fleetid in row.fleetIds]
+        return assets[assets['id'].isin(idx)]
```
[the remark this answers is not in the document]
Comment (delunalara) [reply emspy/query/aircraft.py:84]: No

File: emspy/query/flight.py [hunk 5]
```diff
@@ -87,64 +134,84 @@ def __get_dbtree(self):
...
+        tree = self._trees['dbtree']
+        if self.searchtype == 'contain':
+            dbs = tree[(tree.nodetype == 'database')
+                       & tree.name.str.contains(_treat_spchar(name), case=False)]
+            # Return database with shortest name
+            self._db_id = dbs.loc[dbs['name'].str.len().idxmin(), 'id']
+        # Exact string matching
+        elif self.searchtype == 'match':
+            self._db_id = tree[(tree.nodetype == 'database') &
+                tree.name.str.match(_treat_spchar(name), case=False)]['id'].values[0]
```
[the remark this answers is not in the document]
Comment (delunalara) [reply emspy/query/flight.py:179]: Yeah, it's used differently based on the context, so I'd just leave it in the individual uses for clarity

File: emspy/query/fltquery.py (hunk 2 above)
Comment (delunalara) [reply emspy/query/fltquery.py:11]: - Added the `deselect` method to drop fields from a query
- Modified `filter` with an `operator` parameter to control if the filters are joined with AND or OR
- Added `remove_filter` method to drop a filter from the query
- Made `__split_expr` method a static method
- Modified `__split_expr` to account for `is null` and `is not null` operators
- Added support for `is null` and `is not null` operators, added these as keys to the `sp_ops` dictionary
- Modified each filter function to account for `is null` and `is not null` operators
